### PR TITLE
Adds Shuttle Station, An Entire Station, But On A Shuttle

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
So what if we took the entire station, and put it on a shuttle, then set the crew free on it.
An extremely experimental map where, the entire crew is placed onto a shuttle packed with mostly everything that the station would normally have, and they are free to explore the map.

Behold, The NTSS Companionship:

image

As the ship is this small however, the map will only be available in rotation under 30 players, as 30 people on this is already pushing it enough, we don't need any more.

## Why It's Good For The Game
When the previous versions of shipmap were testmerged, people said might actually make a good lowpop map, so here you go.

## Shit I Might Need To Do

- [ ] Maybe disable syndie bombs from traitor uplinks on this map

If traitors speedrun bombing the shuttle becomes an actual issue, I will do this one.

## Changelog
:cl:
add: A station map entirely located on a shuttle, the NTSS Companionship, only available with less than 30 players online
/:cl:

### CODE REVIEW

File: code/modules/jobs/map_changes/map_changes.dm [hunk 1]
```diff
@@ -5,3 +5,6 @@
 
 #include "..\..\..\..\_maps\map_files\IceBoxStation\job_changes.dm"
 #undef JOB_MODIFICATION_MAP_NAME
+
+#include "..\..\..\..\_maps\map_files\ShipStation\job_changes.dm"
+#undef JOB_MODIFICATION_MAP_NAME
```
Comment (Fikou) [inline code/modules/jobs/map_changes/map_changes.dm:10]: remove

Comment (Paxilmaniac) [reply code/modules/jobs/map_changes/map_changes.dm:10]: oops missed that one

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -635,3 +635,9 @@
 /datum/map_template/shuttle/snowdin/excavation
 	suffix = "excavation"
 	name = "Snowdin Excavation Elevator"
+
+/datum/map_template/shuttle/shipstation
+	port_id = "station"
+	can_be_bought = FALSE
+	suffix = "ship"
+	name = "NTSS 'Friendship'"
```
Comment (Fikou) [inline code/datums/shuttles.dm:643]:
```suggestion
	name = "NTSS 'Companionship'"
```

File: code/modules/shuttle/shipstation.dm [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	desc = "Used to designate a precise transit location for the NTSS 'Companionship'."
+	shuttleId = "station"
+	lock_override = NONE
+	shuttlePortId = "station_custom"
+	jumpto_ports = list("whiteship_away" = 1, "whiteship_home" = 1, "whiteship_z4" = 1, "whiteship_lavaland" = 1)
+	whitelist_turfs = list(/turf/open/space, /turf/open/floor/plating, /turf/open/lava, /turf/closed/mineral)
+	view_range = 6
+	designate_time = 50
+	x_offset = -9
+	y_offset = 11
```
Comment (Fikou) [inline code/modules/shuttle/shipstation.dm:19]: view_range, x_offset and y_offset need tweaking after the shuttle's been resized, console doesnt render it properly

File: config/maps.txt [hunk 4]
```diff
@@ -39,3 +39,8 @@ endmap
 
 map multiz_debug
 endmap
+
+map shipstation
+	maxplayers 40
+	votable
+endmap
```
Comment (Fikou) [inline config/maps.txt:46]: should be in config before runtime and multiz_debug

File: code/datums/shuttles.dm (hunk 2 above)
Comment (ArcaneMusic) [reply code/datums/shuttles.dm]: Canon to the lore

**Comment by ShibaInuLord** [comment]
pog

**Comment by wesoda25** [comment]
I don’t quite understand, how do you plan to implement this? I could understand if it was some sort of gamemode where a traitor(s?) try and sabotage the flight mission or whatever, but as a lone map...? In general I think it could benefit a lot from some sort of direction. 

As for practical feedback, any reason why you excluded a commons area? And 40 seems like way too much, I feel as if <20 is more its speed.

**Comment by Spepbuthewranglesgithub** [comment]
this looks like a living hell for absolutely any antag other than nuke ops, god have mercy.

**Comment by Qustinnus** [comment]
this would benefit much more from being a normal-sized map and not being a real shuttle but doing fake-transitions like FTL13 does.

**Comment by ATH1909** [comment]
How're latejoiners gonna enter the station? Maybe we could use droppods?

**Comment by ATH1909** [comment]
Also, either remove that coffin from the chapel entirely or move it one square left and one square up so that it doesn't block the cremator's tray.

**Comment by MrDoomBringer** [comment]
cool map

**Comment by Catbugg4** [comment]
I actually love this idea but this is lacking some key features from some departments like Toxins, Cryo and somewhere for the AI to sit

**Comment by GunHog** [comment]
What if the ship had two Z-levels? You could put the more space heavy things like a cargo hold, AI chamber, full telecoms, The Supermatter (or other type of engine), and personal quarters down there.

**Comment by Paxilmaniac** [comment]
Multi-z ships aren't real and they can't hurt you.

**Comment by LemonInTheDark** [comment]
Commit mapmerge https://github.com/tgstation/tgstation/wiki/Map-Merger-2%3A-The-Better-One

**Comment by Qustinnus** [comment]
like i said, if you want to do things like multiz you cant do shuttles.

**Comment by TemporalOroboros** [comment]
We just need multiz shuttles then.

**Comment by Qustinnus** [comment]
or instead of using shuttles you just change the backgrounds.

**Comment by TemporalOroboros** [comment]
You would also need to change the rest of the z-level and the connections between z-levels if you want the station to "move".

**Comment by Ghilker** [comment]
use this shuttle to move between servers, Terry's gonna raid Bagil

**Comment by OverlyGeneric** [comment]
The only solution for lack of space is to add more shuttles. Traffic Jam: The Station.

**Comment by tralezab** [comment]
great idea for events, but in actual practice, one traitor can bomb this whole thing sky high. it needs to be external threats only to have a chance.

**Comment by Paxilmaniac** [comment]
I know that traitors and the like completely fuck over the ship, but those are problems I can solve when I'm not as busy.

I would still like if it was testmerged on literally any server just so I can get some feedback for what needs to be changed about the map.

**Comment by Doosteroni** [comment]
Maybe, MAYBE making the cremator publicly accessible is not such a good idea

**Comment by Jared-Fogle** [comment]
Cool as hell. Ping me when this exits the draft stage so I can give it test merge candidate and ask some key holders about this.

**Comment by Jared-Fogle** [comment]
Have you made sure ALL traitor objectives are in, and not just tele?

**Comment by Jared-Fogle** [comment]
Also random question I just thought of, where do nightmares spawn? Are there any antags similar to nightmare that are problematic?

**Comment by Paxilmaniac** [comment]
nightmares and the like spawn in the decently extensive maintenance area on the dock the ship spawns at roundstart

**Comment by MrDoomBringer** [comment]
@Paxilmaniac what if the ship isn't docked when nightmares spawn?

**Comment by Jared-Fogle** [comment]
I wanna see what happens.

**Comment by Qustinnus** [comment]
https://streamable.com/y7hovp If I port this and make it contain cool features like specific sectors having unique things in them, would you use it instead of making the thing an actual shuttle?

**Comment by cacogen** [comment]
cool but probably impossible to get shit done as an antagonist

**Comment by Sikroth** [comment]
Played a very lowpop round of this map on campbell. It was fun for all the wrong reasons, aka trying to work around all the issues and shortcomings was an interesting experience, but if this becomes a regular map they'd just be a major pain.

Issues:
- There's no RnD server, so science doesn't generate points at roundstart until one is built.
- There's an ore silo, but nothing is linked to it.
- Lavaland exists, and you can get there using the teleporter, but there's no way to get back as far as I know (unless you build a quantum pad, take a hand tele or similar).
- Also the lavalands comms relay doesn't work. Probably a side effect of not having an actual comms set onstation? I'm not sure. Maybe having a comms room in the dock and leaving the ship with just a relay would be best. (also the comms mainframe thing was apparently set to NULL, not tcommsat, which might cause issues too)
- Botany doesn't have access to a NutriMax vendor, so no easily acquired nutriments unless they use the biogen carefully. Also no plant bags either (since they also get no lockers with the portable seed extractor), and they need the cargo autolathe for their basic tools.
- I didn't test this myself but apparently the cargo drop pods are extremely inaccurate. Like they can literally fall anywhere on the ship. Not sure if this is an issue that can be solved map wise or if there's something the person using the drop pod console thing was doing wrong.


Suggestions:
- There's no wire to the dock from the ship's SMES's. I think it'd make sense to dock to charge up before leaving, since the dock has an actual engine that is funnily way less probable to kill everyone from rads than the damn Superpacmans.
- On that topic, the ship uses like <40kW. Why use the superpacmans? Get 2 normal ones that don't rad poison everyone forever and generate 40kW unupgraded anyway at min settings. Plus plasma is more plentiful.
- The entire atmos system is a couple T1 canisters. Could use at least a higher tier. Also I didn't check IC but from the screenshot they look like oxygen canisters, not air ones. Probably should use air, or an Oxy+N2 with a mixer like the dock does.
- Xenobio is painful to do. I understand that this is always going to be the case with a map the size of 2 screens though but I wish it was looked into making it less ass. I had to switch around everything to build a little killpen with a HE pipe below for cooling, plus a canister and freezer (the killpen also acted as the RnD server's location because efficiency), along with a xenobio computer because this is 2020 and having to open the killpens and manually toss in monkeys is just a no. And there's also no smartfridge for the cores so it's like the whole thing is just discouraging you from even doing xenobio.
- Funnily though, despite all that, you can technically use the entire ship for xenobio, since it's all 1 area. So you can rain slimes all over the place. I'm fine with this as you can just use any unused areas for makeshift extra xenobio space but maybe it could be an issue?

**Comment by sunofang** [comment]
Cute shuttle! Merge this already

**Comment by Dennok** [comment]
> I didn't test this myself but apparently the cargo drop pods are extremely inaccurate. Like they can literally fall anywhere on the ship. Not sure if this is an issue that can be solved map wise or if there's something the person using the drop pod console thing was doing wrong.

Person that use cargo console can switch between area targeting and beacon targeting. If all shuttle has one area, area targeted pods can drop on any tile of shuttle, this can be solved by mapping or use of beacons.

About lavaland. This shuttle cant land on lavaland?

**Comment by Dennok** [comment]
> https://streamable.com/y7hovp If I port this and make it contain cool features like specific sectors having unique things in them, would you use it instead of making the thing an actual shuttle?

This a great thing but i more like previous version with map and jump radius.
I know that it has issues related to returning to visited sites, but we can say that when we jump we shifted in time and all what out of ship in time of jump lost behind it years. 
And just create new zone with previus seed.

**Comment by Paxilmaniac** [comment]
The shuttle can go to lavaland and deep space, as well as anywhere on the station z-level. People just don't know how to use the shuttle's navigation computer.

**Comment by Paxilmaniac** [comment]
I have made a grave mistake

**Comment by Paxilmaniac** [comment]
Ok actually I'm going to close this and make a new pr in like a week, this needs a lot of work right now

**Comment by stylemistake** [comment]
> and make a new pr

Reeee, just fix this pr and force push.

**Comment by MrDoomBringer** [comment]
@Sikroth 
> I didn't test this myself but apparently the cargo drop pods are extremely inaccurate. Like they can literally fall anywhere on the ship. Not sure if this is an issue that can be solved map wise or if there's something the person using the drop pod console thing was doing wrong.

Person was using it wrong. The map starts with a free cargo pod beacon that you can link with the console to determine exactly where pods will fall.

**Comment by msgerbs** [comment]
@stylemistake I'm pretty sure a force push makes a PR force closed and unable to be reopened.

**Comment by stylemistake** [comment]
No, why?

**Comment by msgerbs** [comment]
I'm almost positive last time I tried that it force closed the PR and would not allow me to reopen it because it had been force pushed.

**Comment by MrDoomBringer** [comment]
msgerbs is right, but have you force pushed since closing it? If not just open it then forcepush your change

**Comment by msgerbs** [comment]
That won't work, it will close it when you force push. GitHub does not allow merging forcepushed PRs because it changes the git history.

**Comment by MrDoomBringer** [comment]
??? lmao that's completely wrong. I forcepush changes to my PR'd branches all the time. Most people rebase that way lol

**Comment by Qustinnus** [comment]
it wont close your pr

**Comment by Qustinnus** [comment]
image

**Comment by msgerbs** [comment]
Idk what the deal is then but this PR: https://github.com/tgstation/tgstation/pull/53196 autoclosed when I force-pushed and the reopen button is grayed out saying "This PR was force-pushed or recreated"

**Comment by stylemistake** [comment]
I don't see any force pushes logged on that PR, only that the branch was deleted and restored, which might the part of the problem.

**Comment by msgerbs** [comment]
Ah, maybe that's it. Weird that the hovertext mentions force-pushing as well though. Oh well, I'll stop spamming poor Paxil's PR about it.

**Pull request closed without merging**